src/modules/twoslash.ts

```diff
@@ -152,18 +152,44 @@ export class TwoslashModule extends Module {
 					e => '// ' + e.renderedMessage.split('\n').join('\n// '),
 				);
 
-				//  ^^^^^ ^^^^^
-				// hats to indicate what token is causing the issue
-				let linkWithHats = '';
-				lineErrors.forEach(e => {
-					if (!e.character) return;
-					const spaceBefore = e.character - linkWithHats.length;
-					linkWithHats += ' '.repeat(spaceBefore);
-					linkWithHats += '^'.repeat(e.length || 0);
-				});
-
-				if (linkWithHats.length > 0) {
-					resultLines.push('//' + linkWithHats.substr(2));
+				// Points to errors, e.g. '       ^^^^   ^^^^^'
+				const hats = lineErrors
+					// only those with a valid span
+					.filter(x => x.character != null && x.length != null)
+					// map to [start, end (non-inclusive)]
+					.map(
+						error =>
+							[
+								error.character!,
+								error.character! + error.length!,
+							] as const,
+					)
+					// sort by start, ascending
+					.sort((a, b) => a[0] - b[0])
```

Comment:
Could this sort be moved before the `map` for a bit more clarity?

Reply:
It could, but more `!`s would be needed